### PR TITLE
Leftnav

### PR DESCRIPTION
**Pull request by eddier**
Adds LeftNav component.  See basic usage example in styleguidist.

This adds a new dependency to the project since it leverages react-stickynode.  Read about it here: https://github.com/yahoo/react-stickynode

Essentially it helps keep the LeftNav in position during scrolling events.

### CODE REVIEW

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereview23676420.surge.sh](http://surgereview23676420.surge.sh)

**Comment by cdaringe** [comment]
@eddier can you baby step us through this? maybe a phone call? we want to learn! 3PM?

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereview23695253.surge.sh](http://surgereview23695253.surge.sh)

**Comment by eddier** [comment]
I added an example implementation to the octo-hacks library that has a working example.  Check that out on the leftnav branch and let me know what questions come up.  Since the leftnav component itself has no react-router knowledge you're left to do the actual navigation in your implementation.

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereview23727828.surge.sh](http://surgereview23727828.surge.sh)

**Comment by cdaringe** [comment]
hmmm, the styleguidist example on leftnav isnt rendering. is it rendering for you?

**Comment by eddier** [comment]
It looks ok on the deploy using Chrome - http://surgereview23727828.surge.sh/#leftnav

Whats your setup?

File: src/components/LeftNav/LeftNav.examples.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+item to active styling.  You can also use an onClick handler to deal with
+navigating the user to the appropriate location.
+
+Manually applied style on this component is only to prevent overrun inside of Styleguide.
+
+#### Supported Fragments ####
+
+```javascript
+`MenuItem`     : Area to display content in the left nav
+```
```
Comment (cdaringe) [inline src/components/LeftNav/LeftNav.examples.md]: - would `LeftNav.propTypes = { children: React.PropTypes.arrayOf(MenuItem) }` or similar work here?
- perhaps we can get the `MenuItem` into the docs so it shows its prop API naturally, vs. a manual listing?

File: src/components/LeftNav/LeftNavMenuItem.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+LeftNavMenuItem.defaultProps = {
+  active: false,
+  image: ''
+}
+
+LeftNavMenuItem.propTypes = {
+  active: React.PropTypes.bool,
+  image: React.PropTypes.string
+}
```
Comment (cdaringe) [inline src/components/LeftNav/LeftNavMenuItem.jsx]: ya, its just seeming like we need a convention to propagate these deeper components' docs up to the docs page

**Comment by cdaringe** [comment]
ah, its back, sorry!  the `maxHeight: '200px'` threw me off a little bit too, but its not really an issue.  i did have some other remarks. let me know what you think?

**Comment by eddier** [comment]
Ok great.  The max height is just there for the docs example.  I'll check out the nested components and see how we can bubble those props up.

**Comment by eddier** [comment]
I did some checking around to see if there is a proper way to deal with this in styleguidist, I found this discussion regarding the nested prop issue:  https://github.com/styleguidist/react-styleguidist/issues/350

**Comment by eddier** [comment]
I wasn't able to find a clean way to bring in the prop types of nested components.  Another option is to add an examples file for the MenuItem component itself.  Chris how do you want to end up handling this?

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereview24059477.surge.sh](http://surgereview24059477.surge.sh)

**Comment by cdaringe** [comment]
how did i miss this too? 

bummer on the findings!

> Chris how do you want to end up handling this?

> "Another option is to add an examples file for the MenuItem component itself."

ya, lets just do _that_ for now then!

thanks @eddier

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereview29768457.surge.sh](http://surgereview29768457.surge.sh)

**Pull request merged**